sql2.txt

```diff
@@ -17,36 +28,71 @@ may have ordered the same product in several different orders.  So the total qua
 SUM(Quantity) AS TotalQuantity and you must GROUP BY CustomerID, ProductID.  Paste your
 SQL statement below.
 
-
+SELECT CustomerName, ProductName, SUM(Quantity) AS TotalQuantity FROM Customers c
+JOIN Orders o
+ON c.CustomerID = o.CustomerID
+JOIN OrderDetails od
+On o.OrderID = od.OrderID
+JOIN Products p
+on od.ProductID = p.ProductID
+GROUP BY c.CustomerID, p.ProductID
+ORDER BY CustomerName ASC;
 
 4. Extend the previous query to give the total price as well as the customer name, product name, 
 and total quantity.  Here you will have
 (SUM(Quantity) * Price) AS TotalPrice.  Paste your SQL statement below.
 
+SELECT CustomerName, ProductName, (SUM(Quantity) * Price) AS TotalPrice FROM Customers c
+JOIN Orders o
+ON c.CustomerID = o.CustomerID
+JOIN OrderDetails od
+On o.OrderID = od.OrderID
+JOIN Products p
+on od.ProductID = p.ProductID
+GROUP BY c.CustomerID, p.ProductID
+ORDER BY CustomerName ASC;
 
 5. Add a new Order to the Orders table.  You will have to specify the ID of an existing customer
 to satisfy the foreign key constraint.  You will also have to specify the ID of an existing
 employee.  Paste your SQL statement below.
 
+INSERT INTO Orders
+(CustomerID, EmployeeID, ShipperID)
```

Comment:
There's no OrderDate here, but since the schema allows it and it's not a requirement in this exercise, I suppose this is fine. It's just a bit strange to have a single order with a `NULL` date.